crates/l1tx/src/filter/indexer.rs

```diff
@@ -23,14 +23,17 @@ pub trait OpsVisitor {
     // Do stuffs with DA.
     fn visit_da<'a>(&mut self, _d: impl Iterator<Item = &'a [u8]>) {}
 
-    fn collect(self) -> Vec<ProtocolOperation>;
+    // Collect data
+    fn collect(
+        self,
+    ) -> (
+        Vec<ProtocolOperation>,
+        Vec<DepositRequestInfo>,
+        Vec<DaEntry>,
+    );
```

Comment:
This is basically an implicit structure, having a complex tuple like this suggests it could be a struct.

Reply:
Yeah, created a one called `L1TxExtract`.